domain/application/service/application.go

```diff
@@ -325,6 +326,18 @@ func (s *ApplicationService) AddUnits(ctx context.Context, name string, units ..
 	return errors.Annotatef(err, "adding units to application %q", name)
 }
 
+// GetUnitLife looks up the life of the specified unit, returning an error
+// satisfying [applicationerrors.UnitNotFoundError] if the unit is not found.
+func (s *ApplicationService) GetUnitLife(ctx context.Context, unitName string) (corelife.Value, error) {
+	var result corelife.Value
+	err := s.st.RunAtomic(ctx, func(ctx domain.AtomicContext) error {
+		unitLife, err := s.st.GetUnitLife(ctx, unitName)
```

Comment:
Do we need to fetch ID first?

Reply:
Eventually, but this code existed well before we settled on that convention. There will need to be work to change a number of methods on state.